pretyping/pretyping.ml

```diff
@@ -1074,8 +1074,11 @@ and pretype_type k0 resolve_tc valcon (env : ExtraEnv.t) evdref lvar = function
 		   evd_comb1 (define_evar_as_sort env.ExtraEnv.env) evdref ev
                | _ -> anomaly (Pp.str "Found a type constraint which is not a type")
            in
-	     { utj_val = v;
-	       utj_type = s }
+           let evkt = e_new_evar env evdref ~src:(loc, knd) ~naming (mkSort s) in
+           let ev,_ = destEvar evkt in
+           evdref := Evd.define ev v !evdref;
+           { utj_val = v;
+	     utj_type = s }
```

Comment:
Perhaps a more precise explanation would be adequat.

1) if the lambda was explicitely typed with an explicit meta, we must keep this meta and instantiate it as it may be used later in the term.
2) Several tools need feedback about how implicit types have been instantiated afterwards, thus we need to keep the binding.